infrastructure/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java

```diff
@@ -0,0 +1,986 @@
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getAllStreamsAddressRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getBootstrapSequencerRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getDefaultSequencerMetricsRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getSequencerTrimRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getStreamsAddressRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getTokenRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getTokenResponse;
+import static org.corfudb.runtime.proto.RpcCommon.SequencerMetricsMsg.SequencerStatus;
+import static org.corfudb.runtime.proto.service.Sequencer.BootstrapSequencerRequestMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.BootstrapSequencerResponseMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.StreamsAddressRequestMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.TokenRequestMsg;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class has test methods that test the RPC calls present in {@link SequencerServer}.
+ * The RequestMsg type messages that are tested are as follows -
+ * - TokenRequestMsg
+ * - BootstrapSequencerRequestMsg
+ * - SequencerTrimRequestMsg
+ * - SequencerMetricsRequestMsg
+ * - StreamsAddressRequestMsg
+ * See test methods for their functionality descriptions.
+ */
+@Slf4j
+public class SequencerServerTest {
+
+    private final AtomicInteger requestCounter = new AtomicInteger();
+
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    // The SequencerServer instance used for testing
+    private SequencerServer sequencerServer;
+
+    // Parameters that need to be mocked
+    @Mock
+    private ServerContext mockServerContext;
+    @Mock
+    private IServerRouter mockServerRouter;
+    @Mock
+    private ChannelHandlerContext mockChannelHandlerContext;
+    @Mock
+    private Map<UUID, StreamAddressSpace> mockStreamAddressSpaceMap;
+    @Mock
+    private Map<UUID, Long> mockStreamTailToGlobalTailMap;
```

Comment:
Out of curiosity, why does this need to be mocked?

Reply:
These two maps were exposed to have some verifications in the test cases, but we now use `new HashMap<>()` pattern in the `SequencerServer` and don't want to go into such details in tests, so they are deleted.